frontend/ui/impl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/table/GitMacheteRepositoryUpdateBackgroundable.java

```diff
@@ -37,19 +38,22 @@ public final class GitMacheteRepositoryUpdateBackgroundable extends Task.Backgro
   private final GitRepository gitRepository;
   private final IBranchLayoutReader branchLayoutReader;
   private final @UI Consumer<@Nullable IGitMacheteRepositorySnapshot> doOnUIThreadWhenDone;
+  private final Consumer<@Nullable IGitMacheteRepository> doOnUIThreadWhenDoneR;
 
   private final IGitMacheteRepositoryCache gitMacheteRepositoryCache;
 
   public GitMacheteRepositoryUpdateBackgroundable(
       Project project,
       GitRepository gitRepository,
       IBranchLayoutReader branchLayoutReader,
-      @UI Consumer<@Nullable IGitMacheteRepositorySnapshot> doOnUIThreadWhenDone) {
+      @UI Consumer<@Nullable IGitMacheteRepositorySnapshot> doOnUIThreadWhenDone,
+      Consumer<@Nullable IGitMacheteRepository> doOnUIThreadWhenDoneR) {
     super(project, getString("action.GitMachete.GitMacheteRepositoryUpdateBackgroundable.task-title"));
 
     this.gitRepository = gitRepository;
     this.branchLayoutReader = branchLayoutReader;
     this.doOnUIThreadWhenDone = doOnUIThreadWhenDone;
+    this.doOnUIThreadWhenDoneR = doOnUIThreadWhenDoneR;
```

Comment:
Also, both fields to be renamed, this `R` prefix is confusing... maybe `doOnUIThreadWhenDoneFor{Repository,Snapshot}`?